src/hardware/configuration.rs

```diff
@@ -486,13 +486,15 @@ pub fn setup(
             .set_speed(hal::gpio::Speed::VeryHigh);
     }
 
-    let mac_addr = match eeprom::read_eui48(&mut eeprom_i2c) {
-        Err(_) => {
-            info!("Could not read EEPROM, using default MAC address");
-            smoltcp::wire::EthernetAddress([0x10, 0xE2, 0xD5, 0x00, 0x03, 0x00])
-        }
-        Ok(raw_mac) => smoltcp::wire::EthernetAddress(raw_mac),
-    };
+    // The EEPROM only gets connected to I2C after the analog supplies are up, which
+    // takes a while (particularly on pre-v1.2 hardware). Wait until a read succeeds,
+    // and then read again to get non-corrupted data.
+    info!("Waiting for I2C EEPROM to become available...");
+    while eeprom::read_eui48(&mut eeprom_i2c).is_err() {}
```

Comment:
Reading in a loop like this may result in a hung application. Maybe a maximum number of attempts would be appropriate before panicking?

Reply:
Yes, having the red LEDs on the front panel after a while might be nicer when running without any debugger/logging. (With the debugger connected, the "Waiting" message would be enough indication.)

Any idea how to best determine the (large) number, though? I'm thinking booting up a board a few times and taking, say, twice or three times the maximum observed value for some safety margin, but I'm really not sure how big the spread would be in practice. (I guess the regulator set pin current would be quite tightly controlled.)

Reply:
The difficulty is that there is no upper bound on when this may succeed. It has been observed that P12V0A may _never_ come up properly on V1.1 hardware. Perhaps we should actually feature gate this to v1.1 stabilizers and have V1.2+ do a single read + unwrap. Given that - what do we _want_ to do if V1.1 has a stuck P12VA0 line? I imagine we should show red LEDs + loop?

Reply:
There are basically three modes: a) Comes up cleanly and quickly (v1.2+). b) Comes up cleanly, but takes a few seconds (v1.1). c) Never comes up (v1.1 due to that bug).

I don't think there is a case where v1.1 comes up only after, say, a minute.

Reply:
Yeah, I would then recommend that we do the following:
1. For V1.2 hardware, read the EUI48 once and unwrap the result. It should never fail.
2. For V1.1, read once per 500ms for ~5 seconds. After that, panic abort and illuminate LEDs.

Since we don't have support for run-time detection of the hardware yet, feel free to omit (1)

Reply:
Agreed.